src/main/java/com/artipie/security/policy/CachedYamlPolicy.java

```diff
@@ -363,16 +364,20 @@ private static PermissionCollection perms(final YamlMapping yaml) {
         /**
          * Get user roles collection.
          * @param yaml Yaml to read roles from
+         * @param user Authenticated user
          * @return Roles collection
          */
-        private static Collection<String> roles(final YamlMapping yaml) {
-            Set<String> roles = Collections.emptySet();
+        private static Collection<String> roles(final YamlMapping yaml, final AuthUser user) {
+            Set<String> roles = new HashSet<>(1);
```

Comment:
We can create one `Set` less

```suggestion
            Set<String> roles = Collections.emptySet();
```